### ISSUE TITLE
75-80% CPU-Usage (starting with 1.9.0beta1)

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#357]
I have 2 tracks playing.
When I click on the waveform of the prelistening track playback freezes for a short moment.

Latency is 42ms.

Have Win7 x64 and installed the beta without deinstalling the last stable version.

**Comment by Holzhaus** [comment]
(by pegasus-renegadetech)
All playback including the master out freezes or just the headphone out?

Can you use the Record Mix feature and let us know if the dropout is heard in that recording as well?

**Comment by Holzhaus** [comment]
(by toomuch)
Master out freezes.

I recorded to WAV and had no clicks or freezes.

**Comment by Holzhaus** [comment]
(by jus)
Is this still a issue with Mixxx 1.9.0 final?

**Comment by Holzhaus** [comment]
(by toomuch)
It has now (Mixxx 1.10.0beta1) dropped to 50-70%. It still think this is a bit high. What CPU load do you have on your systems (preferebly laptops for better comparison)?

**Comment by Holzhaus** [comment]
(by rryan)
Lenovo T400 Core 2 Duo at 2.5GHz with Intel Mobile 4 Series graphics.

25% with 2 tracks playing a 2ms latency
30% with 2 tracks and 4 samplers playing at 2ms latency

**Comment by Holzhaus** [comment]
(by rryan)
toomuch -- what speed is your CPU? 

Can you compare the CPU usage of other DJ software? Maybe download a VirtualDJ trial and see what CPU that uses? 

Mixing audio tracks takes a lot of CPU -- there are still some areas of waste in our engine code that we need to improve, though. Particularly the equalizers.

**Comment by Holzhaus** [comment]
(by toomuch)
CPU Speed is 1.87GHz.

I just saw that even in idle mode (no track playing, no lib scanning, etc.) the CPU load was 45-50%. :O This is not permanent, but just happened as I wrote this message and one more time before.

Tested with Virtual DJ Home 7.0.5 and had a CPU load of about 15-20% with two tracks playing (latency 11ms = playing smoothly).
In Mixxx I have to use 10.7ms latency to hear anything. Before that the track is crawling and cracking.  With ASIO the latency can be dropped to nearly 3ms.

**Comment by Holzhaus** [comment]
(by rryan)
What is your CPU type? Atom ? Core2Duo ? Pentium D? Pentium 4? Celeron ?
 etc.

Thanks
RJ

**Comment by Holzhaus** [comment]
(by toomuch)
I have a Intel Pentium Dual Core T2390 with 3GB RAM (Asus X51L laptop).

**Comment by Holzhaus** [comment]
(by toomuch)
In idle mode (no track loading or playing) I have 50% CPU load.

**Comment by Holzhaus** [comment]
(by rryan)
With waveforms turned on or off?

**Comment by Holzhaus** [comment]
(by toomuch)
Before: Waveforms are enabled, but no track is loaded.

I just wanted to reproduce: Now < 5%. Strange!
Now I loaded a track (70-80% CPU), afterwards stayed at 30% CPU, although I did not play it or something. Just loading the track (13MB, 5:43 min, 320kBit/s) keeps Mixxx consuming 30% CPU constantly. Playing it adds another 20% CPU (pausing it reduces 20% again as expected). Turned waveforms of -> drops CPU below 5%.
Loading a track with no waveforms consumes about 50%, afterwards < 5%. Turning waveforms back on: about 50%.
Just to have said it clearly: No track was playing when I did these tests. Only loading, idling and changing waveforms setting.

Seems like it is connected to this bug #888817.

What is different on my setup that other windows users don't have? :(
